### ISSUE TITLE
A few issues with guards in following mode

### ISSUE THREAD

**Issue by warning-djkajslh** [ldtteam/minecolonies#7450]
### Prerequisites 

- [X] I am running the latest **alpha** version of MineColonies and Structurize for my Minecraft version.
- [X] I checked the MineColonies/Structurize [wiki](https://wiki.minecolonies.ldtteam.com/) and made sure my issue is not covered there.
- [X] I made sure that **this issue is not a duplicate of any existing issue.**

### Context 

- Minecraft Version:
- MineColonies Version:
- Structurize Version:
- Related Mods and their Versions:

## Expected behavior

+ Guards would immediately go after any mobs that hit you
+ Guards would pander around a distance around you
+ Guards would immediately teleport after you

## Actual behavior

- Guards don't attack mobs that hit you, you would have to hit the mob first.
- Guards got carried away too much with auto-targeting mobs and eventually stray too far from you, even in tight grouping
- Guards take awhile to decide to teleport to you from distances away, it is maybe due to their excessive auto-targeting mechanic
- If the mob is not on the list of the guards, or isn't a hostile mobs(i.e:I/F dragons.) then guards will completely ignore when they hit you or you hit them.

## Steps to reproduce the issue

1. Select tight grouping following mode
2. Lead your guard(s) into the mob battlefield at night 
3. Observe the guard mechanic with the issues listed
One time I was spited by a Hydra from a distance and the guards were just ignoring it, if I remember correctly then even after I punched the Hydra, guards still don't attack it despite right next to it, just because at that time Hydra wasn't exist on their mob list

### Logs

- latest.log: 
- crashlog: 

### Notes

Basically I want the guards when following to be as loyal as Minecraft wolves, when following they should switch to protecting their leader instead of citizens.

---
## Viewers

* Add a thumbs-up to the bug report if you are also affected. This helps the bug report become more visible to the team and doesn't clutter the comments.
* Add a comment if you have any insights or background information that isn't already part of the conversation.
